plutarch-test/src/Plutarch/Test/Deterministic.hs

```diff
@@ -0,0 +1,68 @@
+{-# LANGUAGE ImpredicativeTypes #-}
+
+module Plutarch.Test.Deterministic (compileD) where
+
+import qualified Data.Text as T
+import Data.Void (Void)
+import Plutarch (ClosedTerm, compile)
+import qualified Plutus.V1.Ledger.Scripts as Scripts
+import PlutusCore.Default (
+  DefaultFun (Trace),
+  DefaultUni (DefaultUniString),
+  Some (Some),
+  ValueOf (ValueOf),
+  someValueOf,
+ )
+import Replace.Megaparsec (streamEdit)
+import qualified Text.Megaparsec as M
+import qualified Text.Megaparsec.Char as M
+import UntypedPlutusCore (
+  Program (Program),
+  Term (Apply, Builtin, Constant, Delay, Force, LamAbs),
+ )
+import qualified UntypedPlutusCore as UPLC
+
+{- Like `compile`, but the result is deterministic -}
+compileD :: ClosedTerm a -> Scripts.Script
+compileD p = rewriteTraces $ compile p
+
+{- Rewrite the string used by `Trace` so that the script becomes deterministic. -}
+rewriteTraces :: Scripts.Script -> Scripts.Script
+rewriteTraces =
+  walkScript $ \term -> do
+    -- Replace the 's' in `trace s`.
+    Apply () b@(Force _ (Builtin _ Trace)) (Constant () (Some (ValueOf DefaultUniString s))) <- pure term
+    let s' = T.pack . streamEdit ghcPatternMatchReplacement id . T.unpack $ s
```

Comment:
You don't need to use `streamEdit` since it's always in the beginning AFAIK.

Reply:
True; that's an accidental artifact from printTerm rewrite commit. Removed now.